neuralmonkey/trainers/generic_trainer.py

```diff
@@ -39,8 +45,23 @@ def __init__(self, objectives: List[Objective],
         # Flatten the list of lists
         self.var_list = [var for var_list in var_lists for var in var_list]
 
+        if decay_function is None:
+            decay_function = constant_decay()
+
         with tf.name_scope("trainer"):
-            self.optimizer = optimizer or tf.train.AdamOptimizer(1e-4)
+            if global_step is None:
+                global_step = tf.Variable(
+                    0, trainable=False, name="global_step")
+            self.global_step = global_step
+
+            self.optimizer = optimizer_getter(  # type: ignore
+                self.global_step,
+                decay_function)
+            # TODO: avoid accessing private member
```

Comment:
Tohle TODO bych drze zahodil a neštil bych se dál používat `_lr`, ono se to třeba rozbije s novou verzí tensorflow.